springboot/src/main/java/org/mobilitydata/gtfsvalidator/springboot/GtfsValidatorController.java

```diff
@@ -0,0 +1,356 @@
+
+/**
+ * Runs the validation through a {@code SpringBoot} interface triggered by incoming HTTP requests.
+ * Provides methods to push the validation report to Google Cloud Storage.
+ */
+@RestController
+public class GtfsValidatorController {
+
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  private static final String VALIDATION_REPORT_BUCKET_NAME_ENV_VAR = "VALIDATION_REPORT_BUCKET";
+  private static final String VALIDATION_REPORT_BUCKET_NAME =
+      System.getenv(VALIDATION_REPORT_BUCKET_NAME_ENV_VAR);
+  private static final String DEFAULT_OUTPUT_BASE = "output";
+  private static final String DEFAULT_NUM_THREADS = "8";
+  private static final String DEFAULT_BUCKET_LOCATION = "US";
+  private static final String MESSAGE_JSON_KEY = "message";
+  private static final String STATUS_JSON_KEY = "status";
+  private static final String DEFAULT_COUNTRY_CODE = "ZZ";
+
+  /**
+   * Runs {@code org.mobilitydata.gtfsvalidator.cli.Main.main} with the arguments extracted via the
+   * query parameters. Returns the the {@code ResponseEntity} that contains information about the
+   * response's {@code HttpStatus} and a message (as a {@code String}) that gives more information
+   * about success or failure. Please note that this method requires authentication to be setup
+   * prior to execution i.e. environment variable GOOGLE_APPLICATION_CREDENTIALS AND
+   * VALIDATION_REPORT_BUCKET have to be defined.
+   *
+   * <p>Please check:
+   *
+   * <ul>
+   *   <li><a href=https://cloud.google.com/storage/docs/naming-buckets>this documentation for
+   *       bucket naming guidelines</a>
+   *   <li><a
+   *       href=https://cloud.google.com/docs/authentication/getting-started#setting_the_environment_variable>this
+   *       documentation for authentication procedure</a>
+   * </ul>
+   *
+   * <a href=https://cloud.google.com/storage/docs/json_api/v1/status-codes>Possible HTTP status
+   * codes</a>
+   *
+   * @param output_base Base directory to store the outputs
+   * @param threads Number of threads to use
+   * @param country_code Country code of the feed, e.g., `nl`. It must be a two-letter country code
+   *     (ISO 3166-1 alpha-2)")
+   * @param url The fully qualified URL to download GTFS archive
+   * @param validation_report_name The name of the validation report including .json extension.
+   * @param dataset_id the id of the dataset validated
+   * @param commit_sha the commit SHA
+   * @return the {@code ResponseEntity} that contains information about the response's {@code
+   *     HttpStatus} and a message (as a {@code String}) that gives more information about success
+   *     or failure. <a href=https://cloud.google.com/storage/docs/json_api/v1/status-codes>Possible
+   *     HTTP status codes</a>
+   */
+  @GetMapping(value = "/", produces = "application/json; charset=UTF-8")
+  @ResponseBody
+  public ResponseEntity<HashMap<String, String>> run(
```

Comment:
Indeed I was returning a `Json` here before, I switched to `HashMap` because `ResponseEntity` automatically converts the response body as a json string, so no need to build one from scratch. I used a map to keep the following structure (which seems more clear than the former one):
- current response body structure:

```json
{ 
  "message": "message body",
  "status": "status code"
}
```

- former response body structure:

```json
{ 
  "properties": 
  {
    "message": "message body",
  }
}
```